src/test/compile-fail/feature-gate-fn_must_use.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#[must_use]
+fn need_to_use_it() -> bool { true } //~ WARN `#[must_use]` on functions is experimental
+
+fn main() {
+    // FIXME: feature gates are tidy-required to have a specially named (or
+    // comment-annotated) compile-fail test (which MUST fail), but for
+    // backwards-compatibility reasons, we want `#[must_use]` on functions to
+    // be compilable even if the `fn_must_use` feature is absent.  The error
+    // below (which makes this compile-fail test fail) doesn't have anything to
+    // do with the feature-gating, but pragmatically if awkwardly solves this
+    // dilemma until a superior solution can be devised.
+    == //~ ERROR expected expression
```

Comment:
We also have `#[rustc_error]` used throughout the test suite which may help here?